File: Algorithms/Python/Dijkstra.py
class Dijkstra:
    def __init__(self, vertices):
        self.V = vertices
        self.graph = [[0 for _ in range(vertices)]
                      for _ in range(vertices)]

    def dijkstra(self, source):
        dist = [1e7] * self.V
        dist[source] = 0
        shortest_path_tree = [False] * self.V
        for _ in range(self.V):
            u = self.min_distance(dist, shortest_path_tree)
            shortest_path_tree[u] = True
            for v in range(self.V):
                if self.graph[u][v] > 0 and not shortest_path_tree[v] and dist[v] > dist[u] + self.graph[u][v]:
                    dist[v] = dist[u] + self.graph[u][v]
        self.print_sol(dist)

    def min_distance(self, dist, shortest_path_tree):
        minimum = 1e7  # Acts like infinite distance
        min_index = 0
        for v in range(self.V):
            if dist[v] < minimum and not shortest_path_tree[v]:
                minimum = dist[v]
                min_index = v
        return min_index

    def print_sol(self, dist):
        print("Vertex\tDistance from source")
        for vert in range(self.V):
            print(vert, "\t\t", dist[vert])


my_dijkstra = Dijkstra(9)
my_dijkstra.graph = [[0, 4, 0, 0, 0, 0, 0, 8, 0],
                     [4, 0, 8, 0, 0, 0, 0, 11, 0],
                     [0, 8, 0, 7, 0, 4, 0, 0, 2],
                     [0, 0, 7, 0, 9, 14, 0, 0, 0],
                     [0, 0, 0, 9, 0, 10, 0, 0, 0],
                     [0, 0, 4, 14, 10, 0, 2, 0, 0],
                     [0, 0, 0, 0, 0, 2, 0, 1, 6],
                     [8, 11, 0, 0, 0, 0, 1, 0, 7],
                     [0, 0, 2, 0, 0, 0, 6, 7, 0]
                     ]
my_dijkstra.dijkstra(0)

File: Algorithms/JAVA/src/implementing_algorithms/Dijkstra.java
package implementing_algorithms;

public class Dijkstra {
	public void dijkstra(int[][] graph, int source) {
		int graphLen = graph.length;
		boolean[] visitedVertex = new boolean[graphLen];
		int[] distance = new int[graphLen];
		for (int i = 0; i < graphLen; i++) { // initialize
			visitedVertex[i] = false;
			distance[i] = Integer.MAX_VALUE; // like an infinite distance to start with
		}

		distance[source] = 0; // start here
		for (int i = 0; i < graphLen; i++) {
			int minDistVertex = findMinDistance(distance, visitedVertex);
			visitedVertex[minDistVertex] = true;
			// update the distance of all neighboring vertices
			for (int v = 0; v < graphLen; v++)
				if (!visitedVertex[v] && graph[minDistVertex][v] != 0
						&& (distance[minDistVertex] + graph[minDistVertex][v] < distance[v]))
					distance[v] = distance[minDistVertex] + graph[minDistVertex][v];
		}
		// print distances
		for (int i = 0; i < distance.length; i++)
			System.out.println("Distance from " + source + " to " + i + " is " + distance[i]);
	}

	private int findMinDistance(int[] distance, boolean[] visitedVertex) {
		int minDist = Integer.MAX_VALUE;
		int minDistVertex = -1;
		for (int i = 0; i < distance.length; i++)
			if (!visitedVertex[i] && distance[i] < minDist) { // find the minimum distance
				minDist = distance[i];
				minDistVertex = i;
			}
		return minDistVertex;
	}

	public static void main(String[] args) {
		Dijkstra myDijkstra = new Dijkstra();
		int graph[][] = new int[][] { { 0, 0, 1, 2, 0, 0, 0 }, { 0, 0, 2, 0, 0, 3, 0 }, { 1, 2, 0, 1, 3, 0, 0 },
				{ 2, 0, 1, 0, 0, 0, 1 }, { 0, 0, 3, 0, 0, 2, 0 }, { 0, 3, 0, 0, 2, 0, 1 }, { 0, 0, 0, 1, 0, 1, 0 } };

		myDijkstra.dijkstra(graph, 1);
	}
}